File: bindings/cpp/src/handshake.cpp
#include <cstring>
#include <algorithm>

namespace tls {
namespace handshake {

bool parse_client_hello(const uint8_t* data, size_t len, uint8_t* client_random) {
    if (!data || len < 40) return false;
    std::memcpy(client_random, data + 4, 32);
    return true;
}

void build_server_hello(uint8_t* out, size_t* out_len, const uint8_t* server_random) {
    out[0] = 0x02;
    out[1] = 0x03;
    out[2] = 0x04;
    std::memcpy(out + 3, server_random, 32);
    *out_len = 35;
}

}
}

File: bindings/nim/src/handshake.nim
import ./tls

proc buildServerHello(ctx: TLSContext, data: var seq[byte]): bool =
  data.setLen(35)
  data[0] = 0x02.byte
  data[1] = 0x03.byte
  data[2] = 0x04.byte
  for i in 0..31:
    data[3 + i] = ctx.serverRandom[i]
  return true
